vcls/magento2/default.vcl

```diff
@@ -0,0 +1,229 @@
+vcl 4.1;
+
+import goto;
+import std;
+import ykey;
+
+# For SSL offloading, pass the following header in your proxy server or load balancer: '/* {{ ssl_offloaded_header }} */: https'
+
+backend default none;
+
+probe p1 {
+        .url = "/pub/health_check.php";
+        .timeout = 2s;
+        .interval = 5s;
+        .window = 10;
+        .threshold = 5;
+}
+
+acl purge {
+/* {{ ips }} */
+}
+
+sub vcl_init {
+    new dir = goto.dns_director("{{ host }}:{{ port }}",
+                                probe = p1,
+                                first_byte_timeout = 10m);
+}
+
+sub vcl_recv {
+    set req.backend_hint = dir.backend();
+
+    if (req.method == "PURGE") {
+        if (client.ip !~ purge) {
+            return (synth(405, "Method not allowed"));
+        }
+        # To use the X-Pool header for purging varnish during automated deployments, make sure the X-Pool header
+        # has been added to the response in your backend server config. This is used, for example, by the
+        # capistrano-magento2 gem for purging old content from varnish during it's deploy routine.
+        if (!req.http.X-Magento-Tags-Pattern && !req.http.X-Pool) {
+            return (synth(400, "X-Magento-Tags-Pattern or X-Pool header required"));
+        }
+        if (req.http.X-Magento-Tags-Pattern) {
+            set req.http.X-Magento-Tags-Pattern = regsuball(req.http.X-Magento-Tags-Pattern, "\(\(\^\|\,\)", " ");
+            set req.http.X-Magento-Tags-Pattern = regsuball(req.http.X-Magento-Tags-Pattern, "\(\,\|\$\)\)", "");
+            set req.http.X-Magento-Tags-Pattern = regsuball(req.http.X-Magento-Tags-Pattern, "\|", ",");
+            ykey.purge_keys(req.http.X-Magento-Tags-Pattern);
+        }
+        if (req.http.X-Pool) {
+            set req.http.X-Pool = regsuball(req.http.X-Pool, "\(\(\^\|\,\)", " ");
+            set req.http.X-Pool = regsuball(req.http.X-Pool, "\(\,\|\$\)\)", "");
+            set req.http.X-Pool = regsuball(req.http.X-Pool, "\|", ",");
```

Comment:
Doesn’t ykey have support for these kinds of delimiters?

Reply:
PR welcome